xclim/sdba/utils.py

```diff
@@ -462,3 +464,47 @@ def _interp_quantiles_2D(newx, newg, oldx, oldy, oldg):
         dask="parallelized",
         output_dtypes=[np.float],
     )
+
+
+def rank(da, dim="time", pct=False):
```

Comment:
Can this be submitted as a PR to xarray ?

Reply:
Maybe. I don't think I covered all xarray cases, but that shouldn't be difficult.